File: text-translation-demo/csharp/TextTranslationDemo/UserDictionaryClient.cs
using common;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace TextTranslationDemo
{    
    public class UserDictionaryClient
    {
        private string baseUrl;
        public UserDictionaryClient(string baseUrl)
        {
            this.baseUrl = baseUrl;
        }
        public async Task<UserDictionaryItem[]> GetUserDictionaryAsync(
            ClassiiiUser classiiiUser)
        {
            string url = baseUrl + "/dictionary";

            Dictionary<string, object> headers = HttpUtils.BuildHeaders(classiiiUser, url);            

            var content = await HttpUtils.SendAsync(HttpMethod.Get, url, headers, null);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "", data = new { entries = new UserDictionaryItem[] { } } };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                throw new Exception("get user dictionary failed");

            return serverResp.data.entries;
        }
        public async Task<bool> AddUserDictionaryItemAsync(
            ClassiiiUser classiiiUser, UserDictionaryItem item)
        {
            string url = baseUrl + "/dictionary";

            Dictionary<string, object> headers = HttpUtils.BuildHeaders(classiiiUser, url);

            Dictionary<string, object> body = new Dictionary<string, object>() {
                { "fromLang", item.fromLang },
                { "fromText", item.fromText },
                { "toLang", item.toLang },
                { "toText", item.toText }
            };            

            var content = await HttpUtils.SendAsync(HttpMethod.Post, url, headers, body);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "" };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                throw new Exception("add user dictionary item failed");

            return true;
        }

        public async Task<bool> DeleteUserDictionaryItemAsync(
            ClassiiiUser classiiiUser, int id)
        {
            string url = baseUrl + "/dictionary/"+ id;

            Dictionary<string, object> headers = HttpUtils.BuildHeaders(classiiiUser, url);

            var content = await HttpUtils.SendAsync(HttpMethod.Delete, url, headers, null);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "" };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                throw new Exception("delete user dictionary item failed");

            return true;
        }

        public async Task<bool> UpdateUserDictionaryItemAsync(
            ClassiiiUser classiiiUser, int id, UserDictionaryItem item)
        {
            string url = baseUrl + "/dictionary/" + id;

            Dictionary<string, object> headers = HttpUtils.BuildHeaders(classiiiUser, url);

            Dictionary<string, object> body = new Dictionary<string, object>() {
                { "fromLang", item.fromLang },
                { "fromText", item.fromText },
                { "toLang", item.toLang },
                { "toText", item.toText }
            };

            var content = await HttpUtils.SendAsync(HttpMethod.Put, url, headers, body);

            var byteArray = await content.ReadAsByteArrayAsync();

            var responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length);

            var definition = new { status = "" };

            var serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition);

            if (serverResp.status != ResponseStatus.Success)
                throw new Exception("update user dictionary item failed");

            return true;
        }
    }
}

File: text-translation-demo/vb.net/TextTranslationDemo/UserDictionaryClient.vb
Imports common.common
Imports Newtonsoft.Json
Imports System.Net.Http
Imports System.Text

Namespace TextTranslationDemo
    Public Class UserDictionaryClient
        Private baseUrl As String

        Public Sub New(ByVal baseUrl As String)
            Me.baseUrl = baseUrl
        End Sub

        Public Async Function GetUserDictionaryAsync(ByVal rozettaApiUser As RozettaApiUser) As Task(Of UserDictionaryItem())
            Dim url As String = baseUrl & "/dictionary"
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(rozettaApiUser, url)
            Dim content = Await HttpUtils.SendAsync(HttpMethod.Get, url, headers, Nothing)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = "",
                .data = New With {
                    .entries = New UserDictionaryItem() {}
                }
            }
            Dim serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("get user dictionary failed")
            Return serverResp.data.entries
        End Function

        Public Async Function AddUserDictionaryItemAsync(ByVal rozettaApiUser As RozettaApiUser, ByVal item As UserDictionaryItem) As Task(Of Boolean)
            Dim url As String = baseUrl & "/dictionary"
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(rozettaApiUser, url)
            Dim body As Dictionary(Of String, Object) = New Dictionary(Of String, Object)() From {
                {"fromLang", item.fromLang},
                {"fromText", item.fromText},
                {"toLang", item.toLang},
                {"toText", item.toText}
            }
            Dim content = Await HttpUtils.SendAsync(HttpMethod.Post, url, headers, body)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = ""
            }
            Dim serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("add user dictionary item failed")
            Return True
        End Function

        Public Async Function DeleteUserDictionaryItemAsync(ByVal rozettaApiUser As RozettaApiUser, ByVal id As Integer) As Task(Of Boolean)
            Dim url As String = baseUrl & "/dictionary/" & id
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(rozettaApiUser, url)
            Dim content = Await HttpUtils.SendAsync(HttpMethod.Delete, url, headers, Nothing)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = ""
            }
            Dim serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("delete user dictionary item failed")
            Return True
        End Function

        Public Async Function UpdateUserDictionaryItemAsync(ByVal rozettaApiUser As RozettaApiUser, ByVal id As Integer, ByVal item As UserDictionaryItem) As Task(Of Boolean)
            Dim url As String = baseUrl & "/dictionary/" & id
            Dim headers As Dictionary(Of String, Object) = HttpUtils.BuildHeaders(rozettaApiUser, url)
            Dim body As Dictionary(Of String, Object) = New Dictionary(Of String, Object)() From {
                {"fromLang", item.fromLang},
                {"fromText", item.fromText},
                {"toLang", item.toLang},
                {"toText", item.toText}
            }
            Dim content = Await HttpUtils.SendAsync(HttpMethod.Put, url, headers, body)
            Dim byteArray = Await content.ReadAsByteArrayAsync()
            Dim responseString = Encoding.UTF8.GetString(byteArray, 0, byteArray.Length)
            Dim definition = New With {
                .status = ""
            }
            Dim serverResp = JsonConvert.DeserializeAnonymousType(responseString, definition)
            If Not Equals(serverResp.status, ResponseStatus.Success) Then Throw New Exception("update user dictionary item failed")
            Return True
        End Function
    End Class
End Namespace
